meta/saiserialize.c

```diff
@@ -1172,25 +1186,80 @@ int sai_deserialize_enum_list(
         _In_ const sai_enum_metadata_t *meta,
         _Out_ sai_s32_list_t *list)
 {
-    SAI_META_LOG_WARN("not implemented");
-    return SAI_SERIALIZE_ERROR;
+    const char *buf = buffer;
+    int ret;
+    uint32_t idx;
+
+    EXPECT("{");
+
+    EXPECT_KEY("count");
+
+    EXPECT_CHECK(sai_deserialize_uint32(buf, &list->count), uint32);
+
+    EXPECT_NEXT_KEY("list");
+
+    if (strncmp(buf, "null", 4) == 0)
+    {
+        list->list = NULL;
+
+        buf += 4;
+    }
+    else
+    {
+        list->list = calloc((list->count), sizeof(uint32_t));
+
+        EXPECT("[");
+
+        for (idx = 0; idx < list->count; idx++)
+        {
+            if (idx != 0)
+            {
+                EXPECT(",");
+            }
+
+            EXPECT_QUOTE_CHECK(sai_deserialize_enum(buf, meta, &list->list[idx]), enum);
+        }
+
+        EXPECT("]");
+    }
+
+    EXPECT("}");
+
+    return (int)(buf - buffer);
 }
 
 int sai_serialize_attr_id(
         _Out_ char *buf,
         _In_ const sai_attr_metadata_t *meta,
         _In_ sai_attr_id_t attr_id)
 {
-    strcpy(buf, meta->attridname);
+    if (meta != NULL)
+    {
+        strcpy(buf, meta->attridname);
+        return (int)strlen(buf);
+    }
 
-    return (int)strlen(buf);
+    SAI_META_LOG_WARN("failed to serialize attr_id");
+    return SAI_SERIALIZE_ERROR;
 }
 
 int sai_deserialize_attr_id(
         _In_ const char *buffer,
         _Out_ sai_attr_id_t *attr_id)
 {
-    SAI_META_LOG_WARN("not implemented");
+    const sai_attr_metadata_t *meta;
+    size_t len;
+
+    meta = sai_metadata_get_attr_metadata_by_attr_id_name_ext(buffer);
+
+    if (meta != NULL)
+    {
+        len = strlen(meta->attridname);
+        *attr_id = meta->attrid;
+        return (int)len;
```

Comment:
remove len, parameter and just return strlen(meta->attridname);

Reply:
ACK.